### PR TITLE
Wrap-up YAML config

### PR DESCRIPTION
**Pull request by blazember**
Wrapping up YAML config:

* Shade SnakeYaml-Engine library
* Remove check if the library exists
* Add more tests
* Add `advanced-network` YAML configuration support

With advanced network configuration the content of `hazelcast-fullconfig.[xml|yaml]` moved to `hazelcast-fullconfig-without-network.[xml|yaml]`, without network settings. The network settings are defined in either `hazelcast-fullconfig.[xml|yaml]` or `hazelcast-fullconfig-advanced-network-config.[xml|yaml]`. There is an additional test case verifies the equality of the XML and YAML advanced network fullconfig. The test can be relied upon when doing the review.

### CODE REVIEW

File: hazelcast/src/test/resources/hazelcast-fullconfig-advanced-network-config.yaml [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        enabled: false
+        namespace: MY-KUBERNETES-NAMESPACE
+        service-name: MY-SERVICE-NAME
+        service-label-name: MY-SERVICE-LABEL-NAME
+        service-label-value: MY-SERVICE-LABEL-VALUE
+      eureka:
+        enabled: false
+        self-registration: true
+        namespace: hazelcast
+#       <discovery-strategies>
```
Comment (taburet) [inline hazelcast/src/test/resources/hazelcast-fullconfig-advanced-network-config.yaml]: Should this commented out block be here? If yes, it's probably better to rewrite it in YAML.

Comment (blazember) [reply hazelcast/src/test/resources/hazelcast-fullconfig-advanced-network-config.yaml]: Good catch. Will fix in a separate PR if no other need for updating this.

**Comment by taburet** [comment]
run-lab-run

**Comment by blazember** [comment]
Thanks for the review @taburet @vbekiaris

**Pull request merged**